eureka-protempa-etl/src/main/java/edu/emory/cci/aiw/cvrg/eureka/etl/dao/JpaJobDao.java

```diff
@@ -152,6 +164,15 @@ public JpaJobDao(final Provider<EntityManager> inEMProvider) {
 						root.join(JobEntity_.jobEvents).get(JobEvent_.status),
 						jobFilter.getState()));
 			}
+			LOGGER.debug("Checking for recent.");
+			if (jobFilter.getRecent()) {
+				LOGGER.debug("Get recent job");
+				CriteriaQuery<Date> maxQuery = builder.createQuery(Date.class);
+				Root<JobEntity> rootForMax = maxQuery.from(JobEntity.class);
+				maxQuery.select(builder.greatest(rootForMax.get(JobEntity_.created)));
+				Date maxCreatedDate = entityManager.createQuery(maxQuery).getSingleResult();
```

Comment:
Is this executing a query just to get the max date? This should not be necessary.